### DISCUSSION TITLE
[Bug] When using zipkin receiver, an error occurred: not in gzip format

### DISCUSSION THREAD

**Discussion by ii2day** [apache/skywalking#8268] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

OAP server (apache/skywalking)

### What happened

When using zipkin receiver, an error occurred: not in gzip format 
![企业微信截图_16390153226573](https://user-images.githubusercontent.com/45119426/145320803-e7e5b910-ccbb-499b-85fb-4530cc495e20.png)

### What you expected to happen

no error

### How to reproduce

Use zipkin receiver to receive traces data and skywalking version 8.9.0

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
I think this is zipkin client bug? Declaring this format is in gzip format, but actually it isn't?

**Reply by ii2day** [reply]
> I think this is zipkin client bug? Declaring this format is in gzip format, but actually it isn't?

I don't know. I tried the 8.7.0-es7 version again and it had the same problem. Did I use it incorrectly?

**Reply by wu-sheng** [reply]
I don't know either. Personally, I didn't use Zipkin client for a while, including which version may have this issue. You may be better to submit a question to Zipkin community, to see what they know about this. Maybe native Zipkin server has a better way to deal with it. 

```java
        String headEncoding = request.getHeader("accept-encoding");
        if (headEncoding != null && (headEncoding.indexOf("gzip") != -1)) {
            requestInStream = new GZIPInputStream(request.getInputStream());
        } else {
            requestInStream = request.getInputStream();
        }
```

This is why we are doing, can't see what is going wrong.

**Reply by ii2day** [reply]
> I don't know either. Personally, I didn't use Zipkin client for a while, including which version may have this issue. You may be better to submit a question to Zipkin community, to see what they know about this. Maybe native Zipkin server has a better way to deal with it.
> 
> ```java
>         String headEncoding = request.getHeader("accept-encoding");
>         if (headEncoding != null && (headEncoding.indexOf("gzip") != -1)) {
>             requestInStream = new GZIPInputStream(request.getInputStream());
>         } else {
>             requestInStream = request.getInputStream();
>         }
> ```
> 
> This is why we are doing, can't see what is going wrong.

The native Zipkin server does not have this problem.

**Reply by wu-sheng** [reply]
I know, but we don't check, so don't know why. That is how open source works every day :) We don't know all the things. Anyone who wants to do this, should read source codes from both sides.

**Reply by wu-sheng** [reply]
We just checked some Sleuth could works, but very few ones. If you want this fully working in your env, you need to take more time on this.

**Reply by ii2day** [reply]
> We just checked some Sleuth could works, but very few ones. If you want this fully working in your env, you need to take more time on this.

I tried to use opentelemetry's zipkin export library to export to skywalking zipkin receiver, it failed to return this gzip problem, but I received the call interface to send to skywalking zipkin receiver is OK 
![image](https://user-images.githubusercontent.com/45119426/145345562-cd5605db-b2fb-4b36-bdc4-e9ffc02de83a.png)
![image](https://user-images.githubusercontent.com/45119426/145345607-3090a79e-f495-49f6-b30d-d57e579820c8.png)
